app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -403,6 +414,32 @@ class WireActivityViewModel @Inject constructor(
         globalAppState = globalAppState.copy(updateAppDialog = false)
     }
 
+    fun observePersistentConnectionStatus() {
+        viewModelScope.launch {
+            coreLogic.getGlobalScope().observePersistentWebSocketConnectionStatus().let { result ->
+                when (result) {
+                    is ObservePersistentWebSocketConnectionStatusUseCase.Result.Failure -> {
+                        appLogger.e("Failure while fetching persistent web socket status flow from wire activity")
+                    }
+
+                    is ObservePersistentWebSocketConnectionStatusUseCase.Result.Success -> {
+                        result.persistentWebSocketStatusListFlow.collect { statuses ->
+
+                            if (statuses.any { it.isPersistentWebSocketEnabled }) {
+                                if (!servicesManager.isPersistentWebSocketServiceRunning()) {
+                                    servicesManager.startPersistentWebSocketService()
+                                }
+                            } else {
+                                servicesManager.stopPersistentWebSocketService()
+                            }
+                        }
+                    }
```

Comment:
this specific code related to opening incoming call screen was removed from the previous code, why was that? 🤔 

```
notificationManager.observeNotificationsAndCallsWhileRunning(
    usersToObserve,
    scope
) { call -> openIncomingCall(call.conversationId) }
```

I tried to find it somewhere but it just looks like it was deleted and not added elsewhere

Reply:
it's in `PersistentWebSocketService`